### DISCUSSION TITLE
Dynamic Module Federation with SSR

### DISCUSSION THREAD

**Discussion by ArgV04** [nrwl/nx#21866] [Feature Requests]
_Dynamic Module Federation with SSR is not supported and there are no plans to add it.  As we aren't planning to add support for dynamic module federation SSR support, I'm going to close this issue_

_Originally posted by @Coly010 in https://github.com/nrwl/nx/issues/10887#issuecomment-1428093583_
            
Why is this not planned or is it hard to implement and what are the differences between none dynamic SSR with module federation?

**Comment by Coly010** [comment]
Hi @ArgV04 

Planned work indicates something that we have committed to shipping.
We haven't committed to shipping any form of out-of-the-box support for Dynamic Module Federation with SSR as of yet.

There are key things that we must look at when it comes to planning and committing to a feature.

1. Value Add - What value does the feature bring to the user
2. Design - What is the feature exactly and how do we design support around it
3. Implementation Feasibility - How feasible is it to implement the feature
4. Developer Experience - How can we ensure a great DX with the feature

Dynamic Federation is runtime-based. There have been advancements in the Module Federation sphere itself around runtime-based events and middleware, so it could be more feasible now.

Also, promise-based remotes are supported with Nx now, so this type of support has potential to already be supported.

However, there are a lot of additional nuances around Node Federation (the server-side) which lead to unknowns with Dynamic Federation for SSR.

It's not something that we're just ignoring. We are constantly re-evaluating our Module Federation support, and in particular support around SSR with Module Federation. 

I've converted this to a Discussion, so that it can be tracked better than an Issue that could be closed.
